File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/entity/FriendLink.java
package com.aetherblog.blog.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 友链实体
 * 
 * @ref §6.1 - 核心表结构 (V2 增强版)
 */
@Data
@Entity
@Table(name = "friend_links", indexes = {
    @Index(name = "idx_friend_links_visible", columnList = "visible"),
    @Index(name = "idx_friend_links_sort", columnList = "sort_order")
})
public class FriendLink {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, length = 500)
    private String url;

    /**
     * 友链Logo/头像
     */
    @Column(length = 500)
    private String logo;

    @Column(length = 500)
    private String description;

    // ========== V2 新增字段 ==========

    /**
     * 联系邮箱
     */
    @Column(length = 100)
    private String email;

    /**
     * RSS 订阅地址
     */
    @Column(name = "rss_url", length = 500)
    private String rssUrl;

    /**
     * 主题颜色
     */
    @Column(name = "theme_color", length = 20)
    private String themeColor = "#6366f1";

    /**
     * 是否在线 (定期检测)
     */
    @Column(name = "is_online")
    private Boolean isOnline = true;

    /**
     * 最后检测时间
     */
    @Column(name = "last_check_at")
    private LocalDateTime lastCheckAt;

    // ========== 原有字段 ==========

    @Column(name = "sort_order", nullable = false)
    private Integer sortOrder = 0;

    @Column(nullable = false)
    private Boolean visible = true;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: apps/server-go/internal/dto/friend_link.go
package dto

import "time"

// FriendLinkRequest 是创建或更新友情链接接口的请求体 DTO。
type FriendLinkRequest struct {
	Name        string  `json:"name"        validate:"required,max=100"` // 友链名称，最多 100 个字符（必填）
	URL         string  `json:"url"         validate:"required,max=500"` // 友链目标 URL，最多 500 个字符（必填）
	Logo        *string `json:"logo"`                                    // 友链网站 Logo 图片 URL（可选）
	Description *string `json:"description"`                             // 友链描述或简介（可选）
	Email       *string `json:"email"`                                   // 友链站长联系邮箱（可选）
	RSSUrl      *string `json:"rssUrl"`                                  // 友链网站的 RSS 订阅地址（可选）
	ThemeColor  *string `json:"themeColor"`                              // 友链主题色，十六进制颜色值（可选）
	IsOnline    *bool   `json:"isOnline"`                                // 友链网站是否在线（可选，由系统自动检测或手动设置）
	SortOrder   int     `json:"sortOrder"`                               // 排序权重，数值越小排序越靠前
	Visible     *bool   `json:"visible"`                                 // 是否在前台显示（可选）
}

// FriendLinkVO 是友情链接信息的响应视图对象（View Object）。
type FriendLinkVO struct {
	ID          int64      `json:"id"`          // 友链唯一 ID
	Name        string     `json:"name"`        // 友链名称
	URL         string     `json:"url"`         // 友链目标 URL
	Logo        *string    `json:"logo"`        // 友链网站 Logo 图片 URL（可为空）
	Description *string    `json:"description"` // 友链描述或简介（可为空）
	Email       *string    `json:"email"`       // 友链站长联系邮箱（可为空）
	RSSUrl      *string    `json:"rssUrl"`      // 友链网站的 RSS 订阅地址（可为空）
	ThemeColor  *string    `json:"themeColor"`  // 友链主题色（可为空）
	IsOnline    *bool      `json:"isOnline"`    // 友链网站是否在线（可为空，未检测时为 nil）
	LastCheckAt *time.Time `json:"lastCheckAt"` // 上次在线状态检测时间（可为空）
	SortOrder   int        `json:"sortOrder"`   // 排序权重
	Visible     bool       `json:"visible"`     // 是否在前台显示
	CreatedAt   time.Time  `json:"createdAt"`   // 创建时间
	UpdatedAt   time.Time  `json:"updatedAt"`   // 最后更新时间
}

// BatchDeleteRequest 是批量删除资源接口的请求体 DTO。
type BatchDeleteRequest struct {
	IDs []int64 `json:"ids" validate:"required,min=1"` // 需要批量删除的资源 ID 列表（必填，至少 1 条）
}

// ReorderRequest 是批量重新排序资源接口的请求体 DTO。
type ReorderRequest struct {
	IDs []int64 `json:"ids" validate:"required"` // 按新顺序排列的资源 ID 列表（必填，顺序即为新的排列顺序）
}
